ts/Series/Windbarb/WindbarbSeries.ts

```diff
@@ -0,0 +1,566 @@
+                            'highcharts-color-' +
+                            pick(point.colorIndex, point.series.colorIndex)
+                        );
+                }
+
+                // Position the graphic
+                point.graphic
+                    .attr({
+                        d: this.windArrow(point) as any,
+                        translateX: (plotX as any) + this.options.xOffset,
+                        translateY: (plotY as any) + this.options.yOffset,
+                        rotation: point.direction
+                    });
+
+                if (!this.chart.styledMode) {
+                    point.graphic
+                        .attr(this.pointAttribs(point));
+                }
+
+            } else if (point.graphic) {
+                point.graphic = point.graphic.destroy();
+            }
+
+            // Set the tooltip anchor position
+            point.tooltipPos = [
+                (plotX as any) + this.options.xOffset +
+                    (inverted && !this.onSeries ? shapeOffset : 0),
+                (plotY as any) + this.options.yOffset -
+                    (inverted ?
+                        0 :
+                        shapeOffset + (yAxis.pos as any) - chart.plotTop
+                    )
+            ]; // #6327
+        }, this);
+    }
+
+    // Fade in the arrows on initializing series.
+    public animate(
+        init?: boolean): void {
+        if (init) {
+            (this.markerGroup as any).attr({
+                opacity: 0.01
+            });
+        } else {
+            (this.markerGroup as any).animate({
+                opacity: 1
+            }, animObject(this.options.animation));
+        }
+    }
+
+
+}
+
+interface WindbarbSeries {
+    beaufortFloor: Array<number>;
+    beaufortName: Array<string>;
+    getPlotBox: Highcharts.OnSeriesMixin['getPlotBox'];
+    onSeries: Highcharts.OnSeriesSeries['onSeries'];
+    parallelArrays: Array<string>;
+    pointArrayMap: Array<string>;
+    pointClass: typeof WindbarbPoint;
+    trackerGroups: Array<string>;
+    windArrow(point: WindbarbPoint): (SVGElement|SVGPath);
+
+}
+
+extend(WindbarbSeries.prototype, {
+    pointArrayMap: ['value', 'direction'],
+    parallelArrays: ['x', 'value', 'direction'],
+    beaufortName: ['Calm', 'Light air', 'Light breeze',
+        'Gentle breeze', 'Moderate breeze', 'Fresh breeze',
+        'Strong breeze', 'Near gale', 'Gale', 'Strong gale', 'Storm',
+        'Violent storm', 'Hurricane'],
+    beaufortFloor: [0, 0.3, 1.6, 3.4, 5.5, 8.0, 10.8, 13.9, 17.2, 20.8,
+        24.5, 28.5, 32.7], // @todo dictionary with names?
+    trackerGroups: ['markerGroup'],
+    markerAttribs: function (): undefined {
+        return;
+    } as any,
+    getPlotBox: OnSeriesMixin.getPlotBox,
+    // Don't invert the marker group (#4960)
+    invertGroups: noop as any,
+    // No data extremes for the Y axis
+    getExtremes: (): DataExtremesObject => ({})
```

Comment:
This should go into the class pattern as a regular public function.
`(): DataExtremesObject = ({})` => `function (): DataExtremesObject { return {}; }`